src/envs/behavior.py

```diff
@@ -54,15 +52,16 @@ class BehaviorEnv(BaseEnv):
     def __init__(self) -> None:
         if not _BEHAVIOR_IMPORTED:
             raise ModuleNotFoundError("Behavior is not installed.")
+        # behavior_randomize_init_state will always be False in this
+        # config_file because we are not using their scene samplers.
+        # We are loading pre-computed scenes.
         self._config_file = modify_config_file(
-            os.path.join(igibson.root_path,
-                         CFG.behavior_config_file), CFG.behavior_task_name,
-            CFG.behavior_scene_name, CFG.behavior_randomize_init_state)
+            os.path.join(igibson.root_path, CFG.behavior_config_file),
+            CFG.behavior_task_name, CFG.behavior_scene_name, False)
 
         super().__init__()  # To ensure self._seed is defined.
         self._rng = np.random.default_rng(self._seed)
-        self.set_igibson_behavior_env(self._seed)
-        self.igibson_behavior_env.robots[0].initial_z_offset = 0.7
+        self.set_igibson_behavior_env(task_instance_id=0, seed=self._seed)
         self._type_name_to_type: Dict[str, Type] = {}
         # a unique id for saving and loading each task's state
         self.task_num: int = 0
```

Comment:
i'm getting confused about `self.task_num`. is this supposed to be like a counter of all tasks created, or is it supposed to be the ID of the current environment task or something?

Reply:
`self.task_num` is always a counter of the tasks created. If `behavior_randomize_init_state = True` it will also act as the ID of the current environment instance which also goes from [0...19].